symbology/read-symbols-from-mobile-style-file/src/main/java/com/esri/samples/read_symbols_from_mobile_style_file/ReadSymbolsFromMobileStyleFileController.java

```diff
@@ -0,0 +1,272 @@
+import javafx.scene.input.MouseButton;
+import javafx.scene.input.MouseEvent;
+
+import com.esri.arcgisruntime.concurrent.ListenableFuture;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.view.Graphic;
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.symbology.MultilayerPointSymbol;
+import com.esri.arcgisruntime.symbology.Symbol;
+import com.esri.arcgisruntime.symbology.SymbolStyle;
+import com.esri.arcgisruntime.symbology.SymbolStyleSearchParameters;
+import com.esri.arcgisruntime.symbology.SymbolStyleSearchResult;
+
+public class ReadSymbolsFromMobileStyleFileController {
+
+  @FXML private MapView mapView;
+  @FXML private ListView<SymbolStyleSearchResult> hatSelectionListView;
+  @FXML private ListView<SymbolStyleSearchResult> eyesSelectionListView;
+  @FXML private ListView<SymbolStyleSearchResult> mouthSelectionListView;
+  @FXML private ListView<Integer> colorSelectionListView;
+  @FXML private Slider sizeSlider;
+  @FXML private Label sizeLabel;
+  @FXML private ImageView symbolPreview;
+
+  private GraphicsOverlay graphicsOverlay;
+  private MultilayerPointSymbol faceSymbol;
+  private SymbolStyle emojiStyle;
+
+  @FXML
+  public void initialize() {
+
+    // create a map
+    ArcGISMap map = new ArcGISMap(Basemap.createTopographic());
+    // add the map to the map view
+    mapView.setMap(map);
+
+    // create a graphics overlay and add it to the map
+    graphicsOverlay = new GraphicsOverlay();
+    mapView.getGraphicsOverlays().add(graphicsOverlay);
+
+    // load the available symbols from the style file
+    loadSymbolsFromStyleFile();
+
+    // create a listener that builds the composite symbol when an item from the list view is selected
+    ChangeListener<Object> changeListener = (obs, oldValue, newValue) -> buildCompositeSymbol();
+
+    // create a list of the ListView objects and iterate over it
+    List<ListView<SymbolStyleSearchResult>> listViews = Arrays.asList(hatSelectionListView, eyesSelectionListView, mouthSelectionListView);
+    for (ListView<SymbolStyleSearchResult> listView : listViews) {
+      // add the cell factory to show the symbol within the list view
+      listView.setCellFactory(c -> new SymbolLayerInfoListCell());
+      // add the change listener to rebuild the preview when a selection is made
+      listView.getSelectionModel().selectedItemProperty().addListener(changeListener);
+      // add an empty entry to the list view to allow selecting 'nothing', and make it selected by default
+      listView.getItems().add(null);
+      listView.getSelectionModel().select(0);
+    }
+
+    // add colors to the color selection list view
+    colorSelectionListView.getItems().addAll(0xFFFFFF00, 0xFF00FF00, 0xFFFFB6C1);
```

Comment:
Can we use a `ColorPicker` instead? Perhaps place it between the lists and the slider?

Reply:
That's a great observation, I wasn't aware ColorPicker exists. Added, and in the process there's a slight refactor

Reply:
I wasn't aware ColorPicker existed! Added, and done some refactoring in the process.